src/components/Pagination/PaginationInputOnMobile.tsx

```diff
@@ -0,0 +1,139 @@
+import React, { useEffect, useRef, useState } from 'react'
+import { Box, Text } from 'rebass'
+import styled from 'styled-components'
+
+import useTheme from 'hooks/useTheme'
+
+type Props = {
+  className?: string
+  page: number
+  lastPage: number
+  setPage: (p: number) => void
+}
+
+const isValidNumber = (s: string): boolean => {
+  const numberRegex = /^\d+$/
+  return !!s.match(numberRegex)
+}
```

Comment:
why not use `isNaN()`

Reply:
The string must be a valid integer. isNaN is not enough.
```js
 isNaN('-1.2') == true;
 isNaN('3') == true;
```

Reply:
How about `!isNaN(a) && Number.isInteger(Number(a))`

Reply:
```
['-1.2', '-3', '3', '1.2'].map(a => !isNaN(a) && Number.isInteger(Number(a)))
=> [false, true, true, false]
```

Reply:
anyway, in this case regex is shorter, I prefer regex more =))

Reply:
I don't think "page" is negative, or float.
Is `Number.isInteger(Number(a))` enought for us ?

Reply:
This is to validate user input. We want 1, 2, 3, ... only. In the example above, `3e1` gives `true`.
I think one-line regex is the best so far. 

Btw, `Number.isNaN` should be used instead. Check out https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/isNaN, in some first lines.

Reply:
`isNaN` and `Number.isNaN` are there for different purposes. Not to the latter gonna replace the first one.

Reference: https://stackoverflow.com/a/33164924/8153505

In this case, `isNaN` match our need over `Number.isNaN`